### ISSUE TITLE
Leverage gardener-resource-manager for deploying shoot control plane components

### ISSUE THREAD

**Issue by rfranzke** [gardener/gardener#3485]
**How to categorize this issue?**

/area robustness scalability dev-productivity
/kind enhancement
/priority normal

**What would you like to be added**:
The [gardener/gardener-resource-manager](https://github.com/gardener/gardener-resource-manager) is very powerful when it comes to managing resources inside a Kubernetes cluster. We already use it for deploying the system components and optional addons into shoot clusters. We also use it for deploying extensions and started to use it for deploying seed system components (like etcd-druid, for example).
Let's now also use it for deploying shoot control plane components.

**Why is this needed**:
See #3484 and https://github.com/gardener/gardener-resource-manager/issues/104

**Comment by timuthy** [comment]
Do you already have an idea if and how we should adapt the shoot care operation i.e., control plane health checks to this planned change?
Two options I have in mind:
- Gardener cp health checks only consider conditions of `ManagedResource`s, which has the disadvantage that the Gardener care sync period does not determine the actual health check interval any more (rather `health-sync-period` of GRM).
- Leave the checks as they are today, which has the disadvantage that both Gardenlet and GRM execute health checks but can be turned off by a separate GRM instance in the seed w/ disabled health checks.

**Comment by rfranzke** [comment]
I would rather go for the first approach, i.e., leverage what GRM already does wherever possible. As for your mentioned disadvantage: Do you see a particular problem when the interval is not controllable by gardenlet?

**Comment by timuthy** [comment]
> Do you see a particular problem when the interval is not controllable by gardenlet?

No, not in particular 👍

**Comment by rfranzke** [comment]
This issue was motivated by gardener/gardener-resource-manager#104, however, we decided to not pursue the approach requiring gardenlet to also deploy shoot control plane components via `ManagedResource`s. Additionally, @timebertt and myself had some scalability concerns due to the duplication of the resources (e.g., the monitoring Grafana dashboards stored as `ConfigMap`s are already large and would be replicated in a `Secret` used by a `ManagedResource`). Additionally, it's unclear which consequences would result in having multiple additional `ManagedResource`s per shoot control plane.

/close as we decided to not implement it now or anytime soon as there is no strong need/use-case for it
